src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -268,43 +268,47 @@ private async Task ProcessProjectAsync(Project project, ImmutableArray<ISymbol>
         {
             await _progress.OnFindInDocumentStartedAsync(document, cancellationToken).ConfigureAwait(false);
 
-            SemanticModel? model = null;
             try
             {
-                model = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-
-                // start cache for this semantic model
-                FindReferenceCache.Start(model);
+                // We're doing to do all of our processing of this document at once.  This will necessitate all the
+                // appropriate finders checking this document for hits.  We know that in the initial pass to determine
+                // documents, this document was already considered a strong match (e.g. we know it contains the name of
+                // the symbol being searched for).  As such, we're almost certainly going to have to do semantic checks
+                // to now see if the candidate actually matches the symbol.  This will require syntax and semantics.  So
+                // just grab those once here and hold onto them for the lifetime of this call.
+                var model = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
                 foreach (var symbol in symbols)
                 {
                     var globalAliases = TryGet(symbolToGlobalAliases, symbol);
-                    await ProcessDocumentAsync(document, model, symbol, globalAliases, cancellationToken).ConfigureAwait(false);
+                    await ProcessDocumentAsync(model, symbol, globalAliases).ConfigureAwait(false);
                 }
+
+                GC.KeepAlive(root);
```

Comment:
The SemanticModel doesn't hold it strongly?

Reply:
that's a good question.  let me check on that.

Reply:
you're right.  this is held strongly.  i'll remove.